install/kubernetes/cilium/templates/_helpers.tpl

```diff
@@ -18,12 +18,21 @@ then `include "cilium.image" .Values.image`
 will return `quay.io/cilium/cilium:v1.10.1@abcdefgh`
 */}}
 {{- define "cilium.image" -}}
+{{- if not (kindIs "slice" .) }}
+{{- (fail (printf "required list, but got %q" (kindOf .))) }}
+{{- end }}
+{{- if (ne (len .) 2) }}
+{{- (fail (printf "required list of 2 arguments, but got %d" (len .))) }}
+{{- end }}
+{{- $ := index . 0 }}
+{{- with index . 1 }}
 {{- $digest := (.useDigest | default false) | ternary (printf "@%s" .digest) "" -}}
 {{- if .override -}}
 {{- printf "%s" .override -}}
 {{- else -}}
-{{- printf "%s:%s%s" .repository .tag $digest -}}
+{{- printf "%s/%s:%s%s" $.Values.image.registry .repository .tag $digest -}}
```

Comment:
I think the schema should be either:

```
image:
  repository: string
  tag: string
```
or

```
image:
  registry: string
  name: string
  tag: string
```

where `repository` is `registry+name`. This is aligning with our schemas in general, but it'd be way harder to change on upstream

Reply:
Yes, I'm convinced they won't accept this upstream in this shape. Especially that `.Values.image` is specifically for `cilium-agent` not all images. So we'll have to deal with the naming but I'd like to keep it out of scope of this PR.

```
image:
  repository: string
  tag: string
```

:point_up: this is what they have upstream. So the `repository` part looks like `quay.io/cilium/cilium`. Which is correct. This is the image repository. The problem with docker nomenclature is the registry name is part of the repository name.